ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -103,11 +123,38 @@ public boolean isEnabled() {
         return state == FollowStates.FOLLOW_RUNNING || state == FollowStates.FOLLOW_START;
     }
 
+    public FollowLocationSource getLocationSource() {
+        return mLocationSource;
+    }
+
+    public void setLocationSource(FollowLocationSource source) {
```

Comment:
This will of course remove the ability to change the location source during a running follow via `updateFollowParams()`, but I honestly can't envision a case where someone would want the source of Follow's locations to change. So I think that's OK. Agree?